model/client4.go

```diff
@@ -36,10 +37,15 @@ func NewAPIv4Client(url string) *Client4 {
 }
 
 func BuildErrorResponse(r *http.Response, err *AppError) *Response {
-	if r == nil {
-		return &Response{StatusCode: 0, Error: err}
-	} else {
-		return &Response{StatusCode: r.StatusCode, Error: err}
+	statusCode := 0
+	if r != nil {
+		statusCode = r.StatusCode
+	}
+
+	return &Response{
+		StatusCode: statusCode,
+		Error:      err,
+		Response:   r,
```

Comment:
The response can sometimes be nil. Nervous about passing it back because it opens up this error. Do we really need to?
If we do then can we add some comments warning that it can be nil?